skfs/src/Context.sk

```diff
@@ -1718,7 +1737,7 @@ fun withRegionFold<K, T>(
     !optSaved = Some(newObstack());
     loop {
       saved = optSaved.fromSome();
-      it.next() match {
+      callOnParentObstack(saved, it.next) match {
```

Comment:
If I remember the data structure correctly, an obstack is a linked list of pages, so this shouldn't be an issue if it.next allocates causing a new page to be created? It won't corrupt the child obstack? Pretty sure this is true in native where we just malloc pages, but wasm has a simpler allocation scheme I think that's based on bumping, could that cause a problem?

Reply:
I'm trying to remember the wasm implementation. I don't think this is an issue. It only ever increases the bump right?

Reply:
Yes, It increases the bump or takes a previously freed page in the free table so It don't affect the child obstack.